### ISSUE TITLE
Installation in Windows environment - Microsoft Visual Studio

### ISSUE THREAD

**Issue by wicaksonokk** [LHEEA/HOS-ocean#15]
Hello,

I am trying to install HOS-ocean in Windows environment within the Microsoft Visual Studio. I followed the steps written in this page (https://github.com/LHEEA/HOS-ocean/wiki/Installation), but a lot of errors happened during the building of solution, such as:

error #7002: Error in opening the compiled module file. Check INCLUDE paths. [INITIAL_CONDITION]
error #7002: Error in opening the compiled module file. Check INCLUDE paths. [RANDOM_NUMBERS]
error #6638
error #7976
error #6404
error #6724
and a lot more.

Is there anyone having the same experience? Last year, I tried installation in Linux ubuntu, and there were no significant problems. 

![無題](https://user-images.githubusercontent.com/58922371/70874828-426c5300-1ff7-11ea-8438-bd55797e8272.png)

Cordialement, 
Ardhana

**Comment by trimtrim** [comment]
I installed HOS-OCEAN in windows successfully. Please kindly follow the below steps: 
1.	line 253 change "TWOPI*RAND()" in file initial_condition.f90 to "TWOPI*RAN(iseed)"
2.	change all function name “dfftw_plan_r2r_1d_” to “dfftw_plan_r2r_1d” in file fourier_FFTW.f90
3.	change all function name “dfftw_plan_r2r_2d_” to “dfftw_plan_r2r_2d” in file fourier_FFTW.f90
4.	use MKL fftw wrapper instead of HOS fftw function:
     a)	change “ include ‘fftw3.f03’”to “INCLUDE 'fftw3.f'”
     b)	add FFTW include path (C:\Program Files (x86)\IntelSWTools\compilers_and_libraries\windows\mkl\include\fftw
) in the project include path:

**Comment by gducrozet** [comment]
@trimtrim Thank you for your feedback.

Guillaume.

**Comment by lkk12342001** [comment]
I have similar errors during installation in Windows. 
I also tired to follow the steps provided by trimtrim but I can't find those code and files such as "TWOPIRAND()" in file initial_condition.f90 and “dfftw_plan_r2r_1d_” in file fourier_FFTW.f90

Did I forget to install or include anything?

-------------------------------------------------------------------------------------------------------------
error #7002: Error in opening the compiled module file.  Check INCLUDE paths.   [INITIAL_CONDITION]	(HOS-ocean.f90)  Line: 32
error #7002: Error in opening the compiled module file.  Check INCLUDE paths.   [RANDOM_NUMBERS]	(initial_condition.f90) Line: 36
error #6404: This name does not have a type, and must have an explicit type.   [GETPID]		(random_numbers.f90) Line: 62

error #6285: There is no matching specific subroutine for this generic subroutine call.   [READ_DATUM] (input_post_process.f90) Line:68
error #6404: This name does not have a type, and must have an explicit type.   [I_ANA] [I_CARD]		(input_post_process.f90)	Line:67	
error #6404: This name does not have a type, and must have an explicit type.   [X_MIN] [X_MAX] [Y_MIN]....	(input_post_process.f90)	Line: 73,74,75,....	

error #7002: Error in opening the compiled module file.  Check INCLUDE paths.   [READ_FILES]	[RECONSTRUCTION]	(Post_processing.f90)	Line:29,32	
error #6401: The attributes of this name conflict with those made accessible by a USE statement.   [MODESSPECX] [MODESSPECT]....	(Post_processing.f90)	Line:61
error #6401: The attributes of this name conflict with those made accessible by a USE statement.   [DEPTH_STAR]	[XLEN_STAR]...	(Post_processing.f90)	Line:65	
error #6404: This name does not have a type, and must have an explicit type.   [I_CARD] [I_ANA][FILE_3D]		(Post_processing.f90)	Line:77,77,84...	
Severity	Code	Description	Project	File	Line	Suppression State
error #6724: An allocate/deallocate object must have the ALLOCATABLE or POINTER attribute.   [PHIS] [Y] [ETA]	(Post_processing.f90)	Line:180	
error #6404: This name does not have a type, and must have an explicit type.   [FILE_MOD]		(Post_processing.f90)	Line:191	
error #6414: This PARAMETER constant name is invalid in this context.   [N1O2P1][N2O2P1][M1]...		(Post_processing.f90)	Line:194,195,198...	

error #6404: This name does not have a type, and must have an explicit type.   [N_HDR][FILE_3D]		(read_files.f90)	Line:63,138	

error #6414: This PARAMETER constant name is invalid in this context.   [N1O2P1 (reconstruction.f90)	Line:54	
error #6351: The number of subscripts is incorrect.   [KTH] (reconstruction.f90) Line: 99,117,136...

**Comment by wftfwtf** [comment]
Can't we use CMake to install HOS-Ocean by Visual Studio? Is CMake for UNIX/Linux systems?

**Comment by trimtrim** [comment]
> Can't we use CMake to install HOS-Ocean by Visual Studio? Is CMake for UNIX/Linux systems?

Yes. We can use CMake to install HOS-Ocean. However, I haven't tried yet.

**Comment by wftfwtf** [comment]
@trimtrim Thank you for your reply.

Would you tell us the detail of the environment that you used when you succeeded to install HOS-Ocean in Windows?

> I have similar errors during installation in Windows.
> I also tired to follow the steps provided by trimtrim but I can't find those code and files such as "TWOPIRAND()" in file initial_condition.f90 and “dfftw_plan_r2r_1d_” in file fourier_FFTW.f90

I had almost the same errors as lkk12342001 when I tried to install HOS-Ocean. My environment is like the below:
- Visual Studio 2019
- Windows 10
- ifort
I followed an instruction in the "installation" page. 
https://github.com/LHEEA/HOS-ocean/wiki/Installation

Also, there is no "fourier_FFTW.f90" in the repository of HOS-Ocean. Instead, I have found "fourier_r2c_FFTW3.f90" in the repository. Did you compile a different version of HOS-Ocean?

**Comment by sanchor** [comment]
hello, everyone.
it would be much esier to make the installation in MSYS2 with MinG-W64 tools.
![image](https://user-images.githubusercontent.com/43979110/202640011-34640088-ad5f-476f-9fe7-70426cededed.png)
you may need to install fftw libraries first from the pacman responsi
and then cd to the cmake folder and make the install by the commands follows
`    mkdir -p build && \
    cd build && \
    cmake .. -DCMAKE_BUILD_TYPE=Release -G "MinGW Makefiles" && \
    mingw32-make && \
    mingw32-make test`
I hope this is helpful for anyone.

**Comment by DoubleMiaoLi** [comment]
> 你好
> 
> 我正在尝试在Microsoft Visual Studio的Windows环境中安装HOS-ocean。我按照本页（https://github.com/LHEEA/HOS-ocean/wiki/Installation）中编写的步骤进行操作，但是在构建解决方案的过程中发生了许多错误，例如：
> 
> 错误 #7002：打开编译的模块文件时出错。检查 INCLUDE 路径。[INITIAL_CONDITION]错误 #7002：打开编译的模块文件时出错。检查 INCLUDE 路径。[RANDOM_NUMBERS]错误 #6638、错误 #7976、错误 #6404、错误 #6724 等等。
> 
> 有没有人有同样的经历？去年，我尝试在 Linux ubuntu 中安装，没有出现重大问题。
> 
> ![無題](https://user-images.githubusercontent.com/58922371/70874828-426c5300-1ff7-11ea-8438-bd55797e8272.png)
> 
> Cordialement， 阿达纳

您好，请问相关问题您解决了吗？是否在运行sourse文件的代码前进行一些配置操作呢？
